src/isar/apis/models/start_mission_definition.py

```diff
@@ -130,97 +124,97 @@ def check_for_duplicate_ids(items: Union[List[Task], List[STEPS]]):
         )
 
 
-def generate_steps(task) -> List[STEPS]:
-    steps: List[STEPS] = []
+def create_isar_task(start_mission_task_definition) -> TASKS:
 
-    if task.type == TaskType.Inspection:
-        steps.extend(generate_steps_for_inspection_task(task=task))
-    elif task.type == TaskType.DriveTo:
-        steps.append(generate_steps_for_drive_to_task(task=task))
-    elif task.type == TaskType.Localization:
-        steps.append(generate_steps_for_localization_task(task=task))
-    elif task.type == TaskType.ReturnToHome:
-        steps.append(generate_steps_for_return_to_home_task(task=task))
-    elif task.type == TaskType.Dock:
-        steps.append(generate_steps_for_dock_task())
+    if start_mission_task_definition.type == TaskType.Inspection:
+        return create_inspection_task(start_mission_task_definition)
```

Comment:
Are we missing `elif start_mission_task_definition.type == TaskType.MoveArm:` ?

Reply:
When we move arm in flotilla the endpoint called on isar is different, so this file is never reached and therefore the tasktype move arm should not exist in this context. If a mission with move arm reaches this file, something is wrong and it is probably good to have it throw the exception. Can specifically add a throw expection if this code
elif start_mission_task_definition.type == TaskType.MoveArm:
 is true

Reply:
![image](https://github.com/user-attachments/assets/ccea6461-2d72-42f8-b7a4-5ab1b3efd9ac)

Reply:
![image](https://github.com/user-attachments/assets/12309883-f60d-4ea2-a3d0-0f63eaae002c)

Reply:
Then the current solution is good, resolving converstation